onfleet/onfleet.py

```diff
@@ -218,6 +219,14 @@ def __call__(self, *args, **kwargs):
                             error_cause,
                         )
 
+                if response.status_code == httplib.NOT_FOUND:
+                    raise OnfleetResourceNotFound(
```

Comment:
Design-wise, does it make more sense to special-case every HTTP status code? (Presuming there may be others we want to know about later on...) Or does it make more sense to use the generic OnfleetError but pass the status code through so we can access it in our code? 

My leaning is toward option 2, as it'll work cleanly for a pretty wide variety of use cases without having to special-case everything. It also allows us in our own code to more easily combine information about status code AND the data onfleet returns in deciding what to do.